src/eval.c

```diff
@@ -302,11 +328,64 @@ sds luaCreateFunction(client *c, robj *body) {
         return dictGetKey(de);
     }
 
+    /* Handle shebang header in script code */
+    ssize_t shebang_len = 0;
+    if (!strncmp(body->ptr, "#!", 2)) {
+        int numparts,j;
+        shebang_len = strchr(body->ptr, '\n') - (char*)body->ptr;
```

Comment:
eval "#!" 0 will crash, it's without '\n'.

Reply:
right, thanks, i'll fix.